src/pdfviewer/PDFDocument.cpp

```diff
@@ -4200,13 +4202,14 @@ void PDFDocument::setAutoHideToolbars(bool enabled)
 {
 	setToolbarsVisible(!enabled);
 	actionAutoHideToolbars->setChecked(enabled);
-	// since we want to have the MouseMoveEvent down at the pdfWidget (internally e.g. for magnifier) up to
-	// the window (for toolbar hiding) all widgets in between seem to need MouseTracking enabled. Otherwise
-	// they will swallow the move event.
-	QWidget *w = pdfWidget;
-	while (w) {
-		w->setMouseTracking(enabled);
-		w = w->parentWidget();
+	// When autohiding the toolbars we need the mouseMoveEvent of pdfWidget in order to track user activity.
+	// We also enable/disable the mouse-tracking flag of all ancestors of the widget because otherwise the may
+	// swallow the mouseMoveEvent. However we do not change the mouse-tracking flag of the pdfWidget itself
+	// because it must be always enabled (the widget uses mouseMoveEvent to maintain its internal state).
+	if (pdfWidget) {
+		for (QWidget *w = pdfWidget->parentWidget(); w; w = w->parentWidget()) {
```

Comment:
I'm not sure anymore that this is needed at all or makes sense with the modification. The original comment suggested that we had to enable tracking for all parents to make the pdfWidget tracking work. If we're not changing the tracking status of pdfWidget anymore, we should also not need to do it for the parents.

Can you please check if enabling tracking for parents is necessary at all? If not, this loop can be removed. If so, we should do it once in the PDFWidget constructor.

Reply:
I just tried removing this loop from `PDFDocument::setAutoHideToolbars` and it caused problems when auto-hiding of toolbars is enabled. Without the loop it always shows the toolbars when the mouse is over the pdf viewer. That is it does not auto-hide the toolbars. 

I tried moving the loop into the PDFWidget constructor and changing the parameter to `true`, that is 
```
	// Mouse tracking must be always enabled because we use the mouseMoveEvent to update the
	// widget state (e.g. change the cursor image when hovering over links)
	setMouseTracking(true);
	// When autohiding the toolbars we need the mouseMoveEvent of pdfWidget in order to track user activity.
	// We also enable/disable the mouse-tracking flag of all ancestors of the widget because otherwise the may
	// swallow the mouseMoveEvent. However we do not change the mouse-tracking flag of the pdfWidget itself
	// because it must be always enabled (the widget uses mouseMoveEvent to maintain its internal state).
	for (QWidget *w = parentWidget(); w; w = w->parentWidget()) {
		w->setMouseTracking(true);
	}
```
Then the same problem happened - it just does not auto-hide the toolbars. Thinking about it, this is expected because when we execute the PDFWidget constructor, it does not have any parent yet, so none of its future ancestors get its mouse tracking enabled.

One place where we could move the loop to is into `PDFDocument::init` just after `pdfWidget->setPDFDocument(this);`
because at this point the widget already has its ancestor chain setup and we can enable tracking for them.

However I am not really sure that we want `PDFDocument::mouseMoveEvent` to be called when auto-hide is disabled. After all that method checks if auto-hide is enabled and if it is disabled, the method does nothing.

So I my guess that the current code is OK. What do you think about it?